### PR TITLE
feat: upgrade to vite 4.5.0

### PR DESCRIPTION
**Pull request by himself65**
Document: https://github.com/vitejs/vite/commit/520139c

Upstream: https://github.com/vitejs/vite/pull/14498

### CODE REVIEW

**Comment by dai-shi** [comment]
Oh, nice. They backported it.
Let me merge my PR first.

**Comment by himself65** [comment]
let me resolve the conflict

File: packages/waku/src/lib/builder.ts [hunk 1]
```diff
@@ -61,6 +61,9 @@ const analyzeEntries = async (entriesFile: string) => {
       ),
     ],
     ssr: {
+      resolve: {
+        conditions: ["react-server"],
+      },
```
Comment (dai-shi) [inline packages/waku/src/lib/builder.ts:66]: Do you think we need to keep L69-71??

Comment (himself65) [reply packages/waku/src/lib/builder.ts:66]: I think some non-ssr modules might include some `react-server '-only libraries. I think the bugs we met only happened in the SSR/Server module. if we remove it, I think the client-only code might error now.  (that's how I imagine the logic from Vite)

Not sure, let me try in the production

Comment (himself65) [reply packages/waku/src/lib/builder.ts:66]: Let me add some real-prototype

Comment (himself65) [reply packages/waku/src/lib/builder.ts:66]: Yep, we can

Comment (himself65) [reply packages/waku/src/lib/builder.ts:66]: You can try this demo

https://github.com/himself65/waku/tree/11840a4950d020fefdd53781103f09f5c6a0ca7c/playground/react-server

Not sure if we need add this to the e2e test

File: packages/waku/src/lib/middleware/rsc/worker-impl.ts [hunk 2]
```diff
@@ -140,6 +140,11 @@ const getViteServer = async (command: "dev" | "build" | "start") => {
...
       }),
     ],
+    ssr: {
+      resolve: {
+        externalConditions: ["react-server"],
+      },
+    },
     resolve: {
       conditions: ["react-server"],
     },
```
Comment (dai-shi) [inline packages/waku/src/lib/middleware/rsc/worker-impl.ts:150]: Can we remove this too?

**Comment by dai-shi** [comment]
We have a hack to copy something for patch in the "template" script in packages/create-waku/package.json

File: packages/waku/src/lib/builder.ts (hunk 1 above)
Comment (himself65) [reply packages/waku/src/lib/builder.ts]: I think we need to add some ecosystem-ci to ensure everything ok in the future

**Comment by himself65** [comment]
I squash it to 3 commits to help the rebase merge

**Review by dai-shi — APPROVED** [review]
Looks good to me!!

I will "squash and merge" anyway.

**Comment by dai-shi** [comment]
oops, i failed to resolve conflicts correctly. thanks.

**Pull request merged**